### PR TITLE
Refactor/the node job part 5

### PR DESCRIPTION
**Pull request by Aprogiena**
This episode is about replacing direct date time functions access with access through date time provider. There are some minor improvements here and there as well, but just few.

### CODE REVIEW

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs [hunk 1]
```diff
@@ -465,7 +465,7 @@ public NetworkPeer(NetworkAddress peerAddress, Network network, NetworkPeerConne
             parameters = parameters ?? new NetworkPeerConnectionParameters();
             this.Inbound = false;
             this.Behaviors = new NetworkPeerBehaviorsCollection(this);
-            this.MyVersion = parameters.CreateVersion(peerAddress.Endpoint, network);
+            this.MyVersion = parameters.CreateVersion(peerAddress.Endpoint, network, this.dateTimeProvider.GetTimeOffset());
```
Comment (bokobza) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs]: you're passing GetTimeOffset but the `CreateVersion` method was actually using  `DateTimeOffset.UtcNow` before.
Is that intentional?

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerFactory.cs [hunk 2]
```diff
@@ -133,7 +133,7 @@ public NetworkPeer CreateConnectedNetworkPeer(Network network, IPEndPoint endpoi
         {
             var peerAddress = new NetworkAddress()
             {
-                Time = DateTimeOffset.UtcNow,
+                Time = this.dateTimeProvider.GetTimeOffset(),
```
Comment (bokobza) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerFactory.cs]: UtcNow to TimeOffset, is that intentional?

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs (hunk 1 above)
Comment (Aprogiena) [reply src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs]: Yes, that is another thing, I'd like to get rid completely of DateTimeOffset from FN, but that's different thing than introducing date time provider. So now the idea was to replace every `DateTimeOffset.UtcNow` with `this.dateTimeProvider.GetTimeOffset()` and every `DateTime.Now` or `DateTime.UtcNow` to be replaced with `this.dateTimeProvider.GetUtcTime()`.

So at this point I want to preserve the same types, just use the provider interface for delivering the actual value.

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerFactory.cs (hunk 2 above)
Comment (Aprogiena) [reply src/Stratis.Bitcoin/P2P/Peer/NetworkPeerFactory.cs]: yes, same as before, same time, different delivery. UtcNow - check DTP.GetTimeOffset how it is implemented:

```
        public virtual DateTimeOffset GetTimeOffset()
        {
            return DateTimeOffset.UtcNow;
        }

```

**Comment by bokobza** [comment]
ok mate that's good to go.

**Comment by bokobza** [comment]
oh, let's wait for the builds, at least on appveyor.

**Pull request merged**